design/FY2017/NFP-Enhancement for External Shading Calculation .md

```diff
@@ -0,0 +1,104 @@
+
+Various simulation tools, such as Radiance and Daysim, employing state-of-the-art ray-tracing simulation techniques, can be used to pre-calculate the shading fractions for each exterior building surface. However, in the current implementation, EnergyPlus doesn’t allow overriding exterior shading calculations with external data either by EMS or co-simulation interface. Considering a large amount of information and the complexity of calculating the dynamic shading in EnergyPlus, allowing this external input would enable a huge potential to speed up EnergyPlus in urban-scale simulations. Another benefit is to enable reusing the shading results for parametric runs which usually do not change external shading. Adding this feature to EnergyPlus would also allow OpenStudio to run radiance simulations before the EnergyPlus simulation and reusing the radiance calculations for shading. 
+
+
+# Approaches 
+We would add an object **Shading:CalculationMethod** listed as follows. The object contains a mandatory **Field: External Shading Calculation Method**, which has two options namely **UseSchedules** and **InternalCalculation**. The field indicates whether the external shading is pre-calculated and imported as a schedule. If **UseSchedules** is chosen, the schedule field can be linked to a Schedule object **Schedule:File**. The name of the schedule is defined in the **Field: Surfaces External Shaded Fraction Schedule Name**. The **Schedule:File** object points to a CSV text file that has external shaded fraction scheduled values for each external surface considered in the calculation. The schedule values are listed in each column, and the header of the column points to the name of the corresponding exterior surface that receives the shading. For each column, the text contains 8760-8784 hours of surface external shaded fraction data, which is pre-calculated and stored in the text file. The Schedule object would be set up in the initialization, and the results can be loaded at runtime or overwritten through co-simulation or EMS code. The data pairs listed as extensible variables map each building exterior surface to a column in the schedule CSV file. During initializing, all columns are loaded at one, and each is saved corresponding to an exterior surface, and the data can be used in run time.
```

Comment:
I think the way to do this would be to load up schedules for all surfaces and then just ignore or override schedules for selected surfaces. The dominant use case will be external calculations for all surfaces so the input method should be tailored towards that I would think.

Another aspect of this is that for processing speed purposes the data should be probably be organized row-major with surfaces as columns and time steps as rows, so you can easily peel off the shading fractions for a given time step for all surfaces. That kind of organization is not really compatible with per surface shading-fraction schedule files.